service/controller/resource/deployment/update.go

```diff
@@ -110,6 +157,32 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 				continue
 			}
 
+			// If worker deployment, check that master does not have any prohibited states before updating the worker
+			if tcK8sClient != nil {
```

Comment:
Will be kept, as the logic depends on it and it may not get created above if the TC does not exist yet